src/msg/async/AsyncMessenger.cc

```diff
@@ -273,6 +276,37 @@ class C_handle_reap : public EventCallback {
   }
 };
 
+/*******************
+ * Admin Socket Hook
+ */
+
+AsyncMessengerSocketHook::AsyncMessengerSocketHook(
+    AdminSocket& asok, AsyncMessenger& m, const std::string& name)
+    : m_asok(asok),
+      m_msgr(m),
+      m_name(name),
+      m_command(fmt::format("messenger dump {}", name)) {}
+
+AsyncMessengerSocketHook::~AsyncMessengerSocketHook() {
+  m_asok.unregister_commands(this);
+}
+
+int AsyncMessengerSocketHook::call(
+    std::string_view command, const cmdmap_t& cmdmap, const bufferlist&,
+    Formatter* f, std::ostream& errss, ceph::buffer::list& out) {
+  if (command == m_command) {
+    f->open_object_section("status");
+    f->open_object_section("messenger");
+    f->open_object_section(m_name);
+    m_msgr.dump(f);
+    f->close_section();
```

Comment:
```suggestion
    f->dump_object(m_name, m_msgr);
```

Reply:
The dump_object() doesn't work in this case as it requires the `.dump()` being a const method. It's not const, because it pulls a lock at the beginning. 

To make it const we need to pull the lock in the asok hook, but looking at the rest of the class I think dump() actually makes most sense.

Reply:
I think changing the lock to `mutable` makes sense (and is fairly typical). Then the `dump` method can be `const`.

Reply:
Done, but the last push adds filter/flag args to dump so dump_object isn't an option anymore